src/main/java/com/google/sps/data/Challenge.java

```diff
@@ -0,0 +1,39 @@
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Challenge {
+  private String challenge_type;
+  private String name;
+  private List <String> steps;
+
+  public Challenge(String type, String name, List steps){  
+    this.challenge_type = type;
+    this.name = name;
+    this.steps = new ArrayList <>(steps);
+  }
+
+ /* public boolean isValidChallenge(){
+  
+  } */
+
+  public boolean isEqual(Challenge object) {
+    if (this == object){
+      return true;
+    }else if (object instanceof Challenge){
```

Comment:
`object instancef Challenge` test is unneeded. The method's parameter specification ensures `object`'s type is `Challenge`.